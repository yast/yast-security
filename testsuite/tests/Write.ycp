/*
 * YaST2: Modules testsuite
 *
 * Description:
 *   Testsuite for the security module
 *
 * Authors:
 *   Michal Svec <msvec@suse.cz>
 *
 * $Id$
 *
 * testedfiles: Security.ycp PamSettings.ycp Pam.ycp
 */

{

include "testsuite.ycp";

import "Security";

map<string,string> m = $[
    "CONSOLE_SHUTDOWN"		:	"halt",
    "CWD_IN_ROOT_PATH"		:	"r2",
    "CWD_IN_USER_PATH"		:	"r2s",
    "DISPLAYMANAGER_REMOTE_ACCESS":	"r4",
    "ENCRYPTION"		:	"md5",
    "kernel.sysrq"		:	"1",
    "FAIL_DELAY"		:	"l2",
    "GID_MAX"			:	"l3",
    "GID_MIN"			:	"l4",
    "DISPLAYMANAGER_SHUTDOWN"	:	"r3",
    "PASS_MAX_DAYS"		:	"l7",
    "PASS_MIN_DAYS"		:	"l9",
    "PASS_MIN_LEN"		:	"l10",
    "PASS_WARN_AGE"		:	"l11",
    "PERMISSION_SECURITY"	:	"r5",
    "ROOT_LOGIN_REMOTE"		:	"r6",
    "RUN_UPDATEDB_AS"		:	"r7",
    "UID_MAX"			:	"l12",
    "UID_MIN"			:	"l13",
    "SYS_UID_MAX"		:	"l14",
    "SYS_UID_MIN"		:	"l15",
    "SYS_GID_MAX"		:	"l16",
    "SYS_GID_MIN"		:	"l17",
    "USERADD_CMD"		:	"l18",
    "USERDEL_PRECMD"		:	"l19",
    "USERDEL_POSTCMD"		:	"l20",
    "DISABLE_RESTART_ON_UPDATE" :	"r13",
    "DISABLE_STOP_ON_REMOVAL"	:	"r14",
    "DISPLAYMANAGER_ROOT_LOGIN_REMOTE"	: "r16",
    "DISPLAYMANAGER_XSERVER_TCP_PORT_6000_OPEN"	: "r17",
    "net.ipv4.tcp_syncookies"	:	"9",
    "net.ipv4.ip_forward"	:	"10",
    "net.ipv6.conf.all.forwarding"	:	"11",
    "SYSTOHC"			:	"r12",
    "SYSLOG_ON_NO_ERROR"	:	"r15",
    "SMTPD_LISTEN_REMOTE"	:	"r18",
    "DHCPD_RUN_CHROOTED"	:	"r19",
    "DHCPD_RUN_AS"		:	"r20",
    "HIBERNATE_SYSTEM"		:	"r21"
];

map E	= $[
    "target" : $[
	"bash_output" : $[]
    ]
];
map R = $[
    "sysconfig" : $[
        "displaymanager" : $[
	    "DISPLAYMANAGER"    :       "",
	]
    ],
    "target"    : $[
        // FileUtils::Exists returns true:
        "stat"  : $[ 1:2 ]
    ],
];

Security::Settings = m;
Security::modified = true;
TEST(``(Security::Write()),[R,$[],E],nil);

}
