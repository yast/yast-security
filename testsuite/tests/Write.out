Read	.etc.login_defs.FAIL_DELAY nil
Write	.etc.login_defs.FAIL_DELAY "l2" true
Read	.etc.login_defs.GID_MAX nil
Write	.etc.login_defs.GID_MAX "l3" true
Read	.etc.login_defs.GID_MIN nil
Write	.etc.login_defs.GID_MIN "l4" true
Read	.etc.login_defs.LASTLOG_ENAB nil
Write	.etc.login_defs.LASTLOG_ENAB "l5" true
Read	.etc.login_defs.PASS_MAX_DAYS nil
Write	.etc.login_defs.PASS_MAX_DAYS "l7" true
Read	.etc.login_defs.PASS_MIN_DAYS nil
Write	.etc.login_defs.PASS_MIN_DAYS "l9" true
Read	.etc.login_defs.PASS_WARN_AGE nil
Write	.etc.login_defs.PASS_WARN_AGE "l11" true
Read	.etc.login_defs.UID_MAX nil
Write	.etc.login_defs.UID_MAX "l12" true
Read	.etc.login_defs.UID_MIN nil
Write	.etc.login_defs.UID_MIN "l13" true
Read	.etc.login_defs.SYSTEM_UID_MAX nil
Write	.etc.login_defs.SYSTEM_UID_MAX "l14" true
Read	.etc.login_defs.SYSTEM_UID_MIN nil
Write	.etc.login_defs.SYSTEM_UID_MIN "l15" true
Read	.etc.login_defs.SYSTEM_GID_MAX nil
Write	.etc.login_defs.SYSTEM_GID_MAX "l16" true
Read	.etc.login_defs.SYSTEM_GID_MIN nil
Write	.etc.login_defs.SYSTEM_GID_MIN "l17" true
Read	.etc.login_defs.USERADD_CMD nil
Write	.etc.login_defs.USERADD_CMD "l18" true
Read	.etc.login_defs.USERDEL_PRECMD nil
Write	.etc.login_defs.USERDEL_PRECMD "l19" true
Read	.etc.login_defs.USERDEL_POSTCMD nil
Write	.etc.login_defs.USERDEL_POSTCMD "l20" true
Read	.sysconfig.clock.SYSTOHC nil
Write	.sysconfig.clock.SYSTOHC "r12" true
Read	.sysconfig.cron.SYSLOG_ON_NO_ERROR nil
Write	.sysconfig.cron.SYSLOG_ON_NO_ERROR "r15" true
Read	.sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS nil
Write	.sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS "r4" true
Read	.sysconfig.displaymanager.DISPLAYMANAGER_ROOT_LOGIN_REMOTE nil
Write	.sysconfig.displaymanager.DISPLAYMANAGER_ROOT_LOGIN_REMOTE "r16" true
Read	.sysconfig.displaymanager.DISPLAYMANAGER_XSERVER_TCP_PORT_6000_OPEN nil
Write	.sysconfig.displaymanager.DISPLAYMANAGER_XSERVER_TCP_PORT_6000_OPEN "r17" true
Read	.sysconfig.displaymanager.DISPLAYMANAGER_SHUTDOWN nil
Write	.sysconfig.displaymanager.DISPLAYMANAGER_SHUTDOWN "r3" true
Read	.sysconfig.locate.RUN_UPDATEDB_AS nil
Write	.sysconfig.locate.RUN_UPDATEDB_AS "r7" true
Read	.sysconfig.mail.SMTPD_LISTEN_REMOTE nil
Write	.sysconfig.mail.SMTPD_LISTEN_REMOTE "r18" true
Read	.sysconfig.security.PERMISSION_SECURITY nil
Write	.sysconfig.security.PERMISSION_SECURITY "r5 local" true
Read	.sysconfig.services.DISABLE_RESTART_ON_UPDATE nil
Write	.sysconfig.services.DISABLE_RESTART_ON_UPDATE "r13" true
Read	.sysconfig.services.DISABLE_STOP_ON_REMOVAL nil
Write	.sysconfig.services.DISABLE_STOP_ON_REMOVAL "r14" true
Read	.sysconfig.suseconfig.CWD_IN_ROOT_PATH nil
Write	.sysconfig.suseconfig.CWD_IN_ROOT_PATH "r2" true
Read	.sysconfig.suseconfig.CWD_IN_USER_PATH nil
Write	.sysconfig.suseconfig.CWD_IN_USER_PATH "r2s" true
Write	.etc.login_defs nil true
Write	.sysconfig.clock nil true
Write	.sysconfig.cron nil true
Write	.sysconfig.displaymanager nil true
Write	.sysconfig.locate nil true
Write	.sysconfig.mail nil true
Write	.sysconfig.security nil true
Write	.sysconfig.services nil true
Write	.sysconfig.suseconfig nil true
Execute	.target.bash "ln -s -f /lib/systemd/system/poweroff.target /etc/systemd/system/ctrl-alt-del.target" 0
Read	.etc.default.passwd."CRYPT_FILES" nil
Write	.etc.default.passwd."CRYPT_FILES" "sha512" true
Execute	.target.bash_output "/usr/sbin/pam-config -d --cracklib" $[]
Execute	.target.bash_output "/usr/sbin/pam-config -d --cracklib-minlen" $[]
Execute	.target.bash_output "/usr/sbin/pam-config -d --pwhistory-remember" $[]
Write	.etc.default.passwd nil true
Write	.etc.polkit-default-privs_local."org.freedesktop.upower.hibernate" "r21" true
Read	.etc.sysctl_conf."kernel.sysrq" nil
Write	.etc.sysctl_conf."kernel.sysrq" "1" true
Read	.etc.sysctl_conf."net.ipv4.ip_forward" nil
Write	.etc.sysctl_conf."net.ipv4.ip_forward" "10" true
Read	.etc.sysctl_conf."net.ipv4.tcp_syncookies" nil
Write	.etc.sysctl_conf."net.ipv4.tcp_syncookies" "9" true
Read	.etc.sysctl_conf."net.ipv6.conf.all.forwarding" nil
Write	.etc.sysctl_conf."net.ipv6.conf.all.forwarding" "11" true
Execute	.target.bash "echo 1 > /proc/sys/kernel/sysrq" 0
Execute	.target.bash "/usr/bin/chkstat --system" 0
Execute	.target.bash "/sbin/set_polkit_default_privs" 0
Execute	.target.bash "/etc/init.d/dhcpd restart" 0
Execute	.target.bash "/etc/init.d/dhcpd restart" 0
Execute	.target.bash "(test -e /etc/init.d/sendmail && VERBOSE=false /usr/lib/sendmail.d/update && /etc/init.d/sendmail restart) || (test -e /etc/init.d/postfix && /usr/sbin/SuSEconfig.postfix && /etc/init.d/postfix restart)" 0
Execute	.target.bash "/etc/init.d/boot.clock start" 0
Execute	.target.bash "/etc/init.d/boot.ipconfig start" 0
Execute	.target.bash "/etc/init.d/boot.ipconfig start" 0
Execute	.target.bash "/etc/init.d/boot.ipconfig start" 0
Return	true
